src/main/scala/com/gu/subscriptions/cas/directives/ProxyDirective.scala

```diff
@@ -4,30 +4,59 @@ import akka.actor.ActorSystem
 import akka.io.IO
 import akka.pattern.ask
 import akka.util.Timeout
-import com.amazonaws.regions.{Regions, Region}
+import com.amazonaws.regions.{Region, Regions}
 import com.gu.subscriptions.cas.config.Configuration
 import com.gu.subscriptions.cas.config.Configuration.{proxyHost, proxyPort, proxyScheme}
 import com.gu.subscriptions.cas.monitoring.{RequestMetrics, StatusMetrics}
+import com.gu.subscriptions.cas.service.SubscriptionService.{extractZuoraSubscriptionId, verifySubscriptionExpiration}
 import spray.can.Http
 import spray.http.HttpHeaders.Host
+import spray.http.MediaTypes._
 import spray.http.{HttpRequest, HttpResponse}
-import spray.routing.{Directives, RequestContext}
+import spray.routing.{Directives, RequestContext, Route}
 
 import scala.concurrent.Future
 import scala.concurrent.duration._
 
-trait ProxyDirective extends Directives{
+trait ProxyDirective extends Directives {
 
   implicit val actorSystem: ActorSystem
-  implicit val timeout: Timeout = 1.seconds
 
-  import actorSystem.dispatcher
+  val authRoute = path("auth") (post {
+      respondWithMediaType(`application/json`) { requestContext =>
+        proxyRequest(requestContext)
+      }
+    })
+
+  val subsRoute = path("subs") (post {
+      respondWithMediaType(`application/json`) { requestContext: RequestContext =>
+        import scala.concurrent.ExecutionContext.Implicits.global
+        import com.gu.subscriptions.cas.model.json.ModelJsonProtocol._
+        import spray.json._
+        extractZuoraSubscriptionId(requestContext.request.entity.asString)
```

Comment:
I contained them where they are used. Makes for better refactoring, especially since most of them rely on implicits